tests/kani/Intrinsics/SIMD/Construction/main.rs

```diff
@@ -34,8 +33,8 @@ fn main() {
     }
     {
         // Intrinsic updating
-        let m = unsafe { simd_insert(y, 0, 1) };
-        let n = unsafe { simd_insert(y, 1, 5) };
+        let m = unsafe { simd_insert(y, 0, 1_i64) };
+        let n = unsafe { simd_insert(y, 1, 5_i64) };
```

Comment:
Do we know if this limitation (needing that cast) will cause problems in how it gets used in real code?

Reply:
Hey @tedinski, sorry for the late reply.

Just wanted to make clear that this is just mimicking `rustc`'s current behavior. Some SIMD intrinsics require additional type-checking done by the backend, as in [here](https://github.com/rust-lang/rust/blob/c1a859b25a95999ba276075bbd8e284ea675b41a/compiler/rustc_codegen_llvm/src/intrinsic.rs#L1033-L1056). This would prevent us from accepting that, in principle, cannot be compiled with `rustc` (this was the case for this test).